docs/architecture/raft.rst

```diff
@@ -30,55 +30,12 @@ Note that when you have a two-DC cluster with the same number of nodes in each D
 of the DCs is down.
 **We recommend configuring three DCs per cluster to ensure that the cluster remains available and operational when one DC is down.**
 
-.. _enabling-raft-existing-cluster:
-
-Enabling Raft
----------------
-
-.. note::
-  In ScyllaDB 5.2 and ScyllaDB Enterprise 2023.1 Raft is Generally Available and can be safely used for consistent schema management.
-  It will get enabled by default when you upgrade your cluster to ScyllaDB 5.4 or 2024.1.
-  If needed, you can explicitly prevent it from getting enabled upon upgrade.
-
-  .. only:: opensource
-
-    See :doc:`the upgrade guide from 5.2 to 5.4 </upgrade/index>` for details.
-
-ScyllaDB Open Source 5.2 and later, and ScyllaDB Enterprise 2023.1 and later come equipped with a procedure that can setup Raft-based consistent cluster management in an existing cluster. We refer to this as the **Raft upgrade procedure** (do not confuse with the :doc:`ScyllaDB version upgrade procedure </upgrade/index/>`).
-
-.. warning::
-    Once enabled, Raft cannot be disabled on your cluster. The cluster nodes will fail to restart if you remove the Raft feature.
-
-To enable Raft in an existing cluster, you need to enable the ``consistent_cluster_management`` option in the ``scylla.yaml`` file 
-for **each node** in the cluster: 
-
-#. Ensure that the schema is synchronized in the cluster by executing :doc:`nodetool describecluster </operating-scylla/nodetool-commands/describecluster>` on each node and ensuring that the schema version is the same on all nodes.
-#. Perform a :doc:`rolling restart </operating-scylla/procedures/config-change/rolling-restart/>`, updating the ``scylla.yaml`` file for **each node** in the cluster before restarting it to enable the ``consistent_cluster_management`` option:
-
-    .. code-block:: yaml
-
-       consistent_cluster_management: true
-
-When all the nodes in the cluster and updated and restarted, the cluster will start the **Raft upgrade procedure**.
-**You must then verify** that the Raft upgrade procedure has finished successfully. Refer to the :ref:`next section <verify-raft-procedure>`.
-
-Alternatively, you can enable the ``consistent_cluster_management`` option when you are:
-
-* Performing a rolling upgrade from version 5.1 to 5.2 or version 2022.x to 2023.1 by updating ``scylla.yaml`` before restarting each node. The Raft upgrade procedure will start as soon as the last node was upgraded and restarted. As above, this requires :ref:`verifying <verify-raft-procedure>` that the procedure successfully finishes.
-* Creating a new cluster. This does not use the Raft upgrade procedure; instead, Raft is functioning in the cluster and managing schema right from the start.
-
-Until all nodes are restarted with ``consistent_cluster_management: true``, it is still possible to turn this option back off. Once enabled on every node, it must remain turned on (or the node will refuse to restart).
-
 .. _verify-raft-procedure:
```

Comment:
In that case, I think we should leave the "Enabling Raft" section and keep "Verifying that the Raft upgrade procedure finished successfully" and "Verifying that Raft is enabled" as its subsections. "Enabling Raft" would contain:
- the definition of the Raft upgrade procedure,
- the information that in the current version (5.5) if Raft hasn't already been enabled, it can only be enabled by performing a rolling upgrade from 5.4 and:
  - it will happen automatically,
  - there is no way back.

I don't think we need to keep information about the preconditions like verifying the schema synchronization. They are mentioned in upgrade guides to 5.2 and 5.4 (after merging #16347), and will be mentioned in upgrade guides to 5.5. IIUC, "Enabling Raft" contained these preconditions because they were also necessary when performing the Raft upgrade procedure without the rolling upgrade. In 5.5, the only case, in which the Raft upgrade procedure is performed, is after the rolling upgrade from 5.4. Since we decided to remove all documentation irrelevant in 5.5, I think we can also remove these preconditions from "Raft upgrade procedure".

@kbr-scylla @annastuchlik What do you think about these ideas?

Reply:
Sounds good